review/src/NoDuplicatePorts.elm

```diff
@@ -0,0 +1,114 @@
+module NoDuplicatePorts exposing (rule)
+
+{-| Highlights duplicate port names, which will break the Elm compiler
+-}
+
+import Dict exposing (Dict)
+import Elm.Syntax.Declaration as Declaration exposing (Declaration)
+import Elm.Syntax.Module as Module exposing (Module)
+import Elm.Syntax.ModuleName exposing (ModuleName)
+import Elm.Syntax.Node as Node exposing (Node)
+import Elm.Syntax.Range exposing (Range)
+import Review.Rule as Rule exposing (Direction, Error, Rule)
+
+
+error : { message : String, details : List String }
+error =
+    { message = "Ensure that port names are unique across your project."
```

Comment:
I'd suggest something like: "Another port with the same name already exists.", because this explains the problem quite fast and feels a bit more actionable. Basically, what you put in details should be your message.

Reply:
That makes sense.  I also wasn't sure whether to use the port name in the error, like NoUnused.Variables does.  For example: "Another port named `alarm` already exists."

Reply:
You could. I think that could be useful, since that will make the user not have to read the code extract :+1: